src/clients/add-on_auto.rb

```diff
@@ -197,7 +213,7 @@ def main
                 # just report an error
                 # TRANSLATORS: The placeholders are for the product name and the URL.
                 error_string = format(_("Failed to add product \"%s\" via\n%s."),
-                  prod["product"], media)
+                  prod["product"] || "<not_defined_name>", media)
```

Comment:
This value is placed into a translated string so we should translate it as well:
```ruby
# TRANSLATORS: a fallback string for undefined product name
prod["product"] || _("<undefined product>")
```